apps/src/lib/ui/lti/link/LtiLinkAccountPage/types.ts

```diff
@@ -0,0 +1 @@
+export type LtiProvider = 'Canvas' | 'Schoology';
```

Comment:
I think we actually have to map the values from the keys of [LMS_PLATFORMS in the LTI policy module](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/lib/policies/lti.rb#L50-L79) into these human readable LMS names. Could do that here or just do it in the controller before we redirect to this page. Might be cleaner to do it in the controller so that we only ever end up with user-friendly strings in the frontend.

Example of the component locally with the value `canvas_cloud` which should map to "Canvas"
<img width="448" alt="Screenshot 2024-05-14 at 6 37 33 PM" src="https://github.com/code-dot-org/code-dot-org/assets/131809324/812e6e39-f11c-444e-9b3e-cd653173b16c">

Reply:
Hmm unfortunately I won't be able to pass this up to the TS union type as a dynamic value - not even as a shared constant via the shared constants file. We would probably need to write something that creates a Typescript definition file (similar to the shared constants generator) for this to work correctly. I'm thinking we can potentially roll this into a future API contract discussion where we refactor our APIs to have proper TS definition generation, what do you think?

Reply:
I dug into this a little more I figured out what you mean, I've updated the PR to publish both the raw type and the friendly name from the controller.